paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -531,6 +533,32 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         }
     }
 
+    private fun onPaymentLauncherResult(launcherResult: PaymentLauncherResult) {
```

Comment:
With `onPaymentResult` now only being used by Link, should we switch it over too (and does that even give us any benefit)?

Reply:
From what I can tell, Link only produces the payment method that was used by not a stripe intent. If we can pull the intent, then we can make it use `PaymentLauncherResult`. Otherwise, there are a few things we can do.
- We could limit `PaymentLauncherResult` to just return the payment method then have `Link` use that result inside of `PaymentResult`.
-  Have `Link` return its own response type with the payment method and remove `onPaymentResult`.

Either way, there isn't any huge benefit to doing it since we Link method aren't saved. `onPaymentResult` is a `BaseSheetViewModel` method that is used in other places so we'd have to remove them there as well.